src/app/models/__tests__/workspace.server.model.spec.ts

```diff
@@ -0,0 +1,196 @@
+  _id: MOCK_WORKSPACE_ID,
+  title: 'Workspace1',
+  admin: MOCK_USER_ID,
+  formIds: [],
+}
+
+describe('Workspace Model', () => {
+  let FORM_COLLAB_READ_ONLY_USER: IUserSchema
+  let FORM_COLLAB_READ_AND_WRITE_USER: IUserSchema
+  let FORM_ADMIN_USER: IUserSchema
+
+  beforeAll(async () => await dbHandler.connect())
+  beforeEach(async () => {
+    const { form, user: adminUser } = await dbHandler.insertEncryptForm({
+      formId: MOCK_FORM_ID,
+      userId: MOCK_USER_ID,
+    })
+    const { user: collabReadOnlyUser } =
+      await dbHandler.insertFormCollectionReqs({
+        userId: new ObjectId(),
+        mailName: 'readOnly',
+      })
+    const { user: collabReadAndWriteUser } =
+      await dbHandler.insertFormCollectionReqs({
+        userId: new ObjectId(),
+        mailName: 'readAndWrite',
+      })
+    await form.updateFormCollaborators([
+      {
+        email: collabReadOnlyUser.email,
+        write: false,
+      },
+      {
+        email: collabReadAndWriteUser.email,
+        write: true,
+      },
+    ])
+
+    await Workspace.create(MOCK_WORKSPACE_FIELDS)
+    FORM_ADMIN_USER = adminUser
+    FORM_COLLAB_READ_AND_WRITE_USER = collabReadAndWriteUser
+    FORM_COLLAB_READ_ONLY_USER = collabReadOnlyUser
+  })
+  afterEach(async () => await dbHandler.clearDatabase())
+  afterAll(async () => await dbHandler.closeDatabase())
+
+  describe('Schema', () => {
+    it('should create and save successfully', async () => {
+      const { user: existentUser } = await dbHandler.insertFormCollectionReqs({
+        userId: new ObjectId(),
+        mailName: 'userThatExists',
+      })
+      const expectedWorkspaceObject = {
+        title: 'Workspace2',
+        admin: existentUser._id,
+        formIds: [],
+      }
+      const validWorkspace = new Workspace(expectedWorkspaceObject)
+      const saved = await validWorkspace.save()
+      const actualSavedObject = omit(saved.toObject(), [
+        '_id',
+        'created',
+        'lastModified',
+        '__v',
+      ])
+
+      expect(saved._id).toBeDefined()
+      expect(saved.created).toBeInstanceOf(Date)
+      expect(saved.lastModified).toBeInstanceOf(Date)
+      expect(actualSavedObject).toEqual(expectedWorkspaceObject)
+    })
+
+    it("should update workspace's formIds successfully because user is the form admin", async () => {
+      const expectedWorkspaceObject = {
+        title: 'Workspace2',
+        admin: FORM_ADMIN_USER._id,
+        formIds: [MOCK_FORM_ID],
+      }
+      const validWorkspace = new Workspace(expectedWorkspaceObject)
+      const saved = await validWorkspace.save()
+      const actualSavedObject = omit(saved.toObject(), [
+        '_id',
+        'created',
+        'lastModified',
+        '__v',
+      ])
+
+      expect(actualSavedObject).toEqual(expectedWorkspaceObject)
+    })
+
+    it("should create and save successfully because user has read and write access in the form's permissionList", async () => {
```

Comment:
we normally use "when" rather than "because", since we're talking about different situations rather than cause and effect

Reply:
I see, I'll update the test descriptions, thanks!